### PR TITLE
Modify readme for crypto hashing and resource id replacement

### PR DESCRIPTION
**Pull request by moria97**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,6 +271,7 @@ The elements can be specified using [FHIRPath](http://hl7.org/fhirpath/) syntax.
 |keep|All elements| Retains the value as is. |
 |redact|All elements| Removes the element. See the parameters section below to handle special cases.|
 |dateShift|Elements of type date, dateTime, and instant | Shifts the value using the [Date-shift algorithm](#date-shift-algorithm).
+|cryptoHash|All elements| Transforms the value using [Crypto-hash method](#Crypto-hash-method). |
```
Comment (ranvijaykumar) [inline README.md:274]: Make sure the hyperlink works. I could not verify it.

File: README.md [hunk 2]
```diff
@@ -280,6 +284,7 @@ Parameters affect the de-identification methods specified in the FHIR path rules
 | ----- | ----- | ----- | ----- | ----- | ----- |
 | dateShift |dateShiftKey|date, dateTime, instant fields| string|A randomly generated string|This key is used to generate date-shift amount in the [Date-shift algorithm](#date-shift-algorithm). 
 | dateShift |dateShiftScope|date, dateTime, instant fields| resource, file, folder | resource | This parameter is used to select date-shift scope. Dates within the same scope will be shifted the same amount. Please provide dateShiftKey together with it. |
+| cryptoHash |cryptoHashKey|All hashing fields| string|A randomly generated string|This key is used for HMAC-SHA256 algorithm in [Crypto-hash method](#Crypto-hash-method).
```
Comment (ranvijaykumar) [inline README.md:287]: Please ensure that the hyperlink works.

File: README.md [hunk 3]
```diff
@@ -311,6 +316,11 @@ You can specify dateShift as a de-identification method in the configuration fil
 2. If the input date/dateTime/instant value is indicative of age over 89, it will be redacted (including year) according to HIPAA Safe Harbor Method.
 3. If the input dateTime/instant value contains time, time will be redacted. Time zone will keep unchanged.
 
+## Crypto-hash method
+You can specify the crypto-hash method in the configuration file. We use HMAC-SHA256 algorithm, which outputs a Hex encoded representation of the hashed output (for example, ```a3c024f01cccb3b63457d848b0d2f89c1f744a3d```). Considering anonymization output conformance, we recommend you not to specify crypto hash method for FHIR elements with special format requirements like *date*, *datetime*, *url*, etc.
```
Comment (ranvijaykumar) [inline README.md:320]: Consider changing the last sentence to this: If you want the anonymized output to be conformant to the FHIR specification, use Crypto-hash on only those fields that can take string of length ....

File: README.md [hunk 4]
```diff
@@ -311,6 +316,11 @@ You can specify dateShift as a de-identification method in the configuration fil
...
 3. If the input dateTime/instant value contains time, time will be redacted. Time zone will keep unchanged.
 
+## Crypto-hash method
+You can specify the crypto-hash method in the configuration file. We use HMAC-SHA256 algorithm, which outputs a Hex encoded representation of the hashed output (for example, ```a3c024f01cccb3b63457d848b0d2f89c1f744a3d```). Considering anonymization output conformance, we recommend you not to specify crypto hash method for FHIR elements with special format requirements like *date*, *datetime*, *url*, etc.
+
+A typical scenario is to replace resource ids across FHIR resources via crypto hashing. With a specific hash key, same resource ids that reside in resources and references will be hashed to a same value. There is a special case when crypto hashing a [literal reference](https://www.hl7.org/fhir/references.html#literal) element. We capture and transform only the id part from a reference, for example, reference ```Patient/123``` will be hashed to ```Patient/a3c024f01cccb3b63457d848b0d2f89c1f744a3d```. In this way, you can easily resolve references across anonymized FHIR resources.
+
```
Comment (Patrick2011) [inline README.md:323]: The tool captures and transforms only the id part ......

File: README.md (hunk 1 above)
Comment (moria97) [reply README.md:274]: Sure. I have verified that it's working https://github.com/microsoft/FHIR-Tools-for-Anonymization/tree/personal/yufei/cryptohash-readme#Crypto-hash-method

File: README.md (hunk 2 above)
Comment (moria97) [reply README.md:287]: Sure. It is working.

File: README.md (hunk 4 above)
Comment (moria97) [reply README.md:323]: Fixed, thanks.

File: README.md (hunk 3 above)
Comment (moria97) [reply README.md]: Sure, updated the last sentence. Thanks!

**Pull request merged**